examples/mistral_qlora_finetune_example.py

```diff
@@ -0,0 +1,54 @@
+"""Example of how to fine-tune a model using the QLoraTrainer class."""
+
+import os
+
+from datasets import load_from_disk
+
+from prompt2model.model_trainer.qlora_trainer import QLoraTrainer
+from prompt2model.utils.dataset_utils import format_train_data, make_combined_datasets
+
+if __name__ == "__main__":
+    # load in datasets
+    retrieved_dataset_dict = load_from_disk("demo_retrieved_dataset_dict")
+    retrieved_dataset = retrieved_dataset_dict["train"]
+    generated_dataset = load_from_disk("demo_generated_dataset")
+    dataset_list = [retrieved_dataset, generated_dataset]
+
+    # combine datasets and create train and eval splits
+    train_dataset = make_combined_datasets(dataset_list)
+    splits = train_dataset.train_test_split(test_size=0.1)
+    train_dataset = splits["train"]  # has 2 cols: "input_col" and "output_col"
+    eval_dataset = splits["test"]  # has 2 cols: "input_col" and "output_col"
+    formatted_train_dataset = format_train_data(
+        train_dataset
+    )  # combined into one col: "text"
+    formatted_eval_dataset = format_train_data(
+        eval_dataset
+    )  # combined into one col: "text"
+
+    # set hyperparams
+    num_epochs = 1
+    qlora_alpha = 8
+    qlora_r = 16
+    qlora_lr = 1e-5
+    save_folder_path = "qlora_finetuned_model"
+    load_best_model_at_end = False
+
```

Comment:
[Python's style guidelines suggest using inline comments sparingly](https://peps.python.org/pep-0008/#inline-comments). Use comments in their own line whenever possible (which probably applies to all of these comments).

Also, comments should almost always be full, grammatical English sentences with proper capitalization and punctuation.